src/OpcUaPiXtendServer/PiXtend/pixtend.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+   Copyright 2021 Kai Huebl (kai@huebl-sgh.de)
+
+   Lizenziert gemäß Apache Licence Version 2.0 (die „Lizenz“); Nutzung dieser
+   Datei nur in Übereinstimmung mit der Lizenz erlaubt.
+   Eine Kopie der Lizenz erhalten Sie auf http://www.apache.org/licenses/LICENSE-2.0.
+
+   Sofern nicht gemäß geltendem Recht vorgeschrieben oder schriftlich vereinbart,
+   erfolgt die Bereitstellung der im Rahmen der Lizenz verbreiteten Software OHNE
+   GEWÄHR ODER VORBEHALTE – ganz gleich, ob ausdrücklich oder stillschweigend.
+
+   Informationen über die jeweiligen Bedingungen für Genehmigungen und Einschränkungen
+   im Rahmen der Lizenz finden Sie in der Lizenz.
+
+   Autor: Kai Huebl (kai@huebl-sgh.de)
+   Autor: Samuel Huebl (Samuel@huebl-sgh.de)
+
+   Usage:
+   pixtend <Module> <Pin> <Operation>
+
+   	   Module:		[V2L | V2S | DO | AO]
+   	   Pin:			D0, ...
+   	   Operation:	read |
+   	   				write [<D-Value> | A-Value]
+   	   D-Value:		on | off
+   	   A-Value:		double
+
+ */
+
+#include <boost/program_options.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <iomanip>
+#include <memory>
+
+#include "OpcUaPiXtendServer/PiXtend/PiXtendV2S.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendV2SDummy.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendV2SInst.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendV2L.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendV2LDummy.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendV2LInst.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendEIOAO.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendEIOAODummy.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendEIOAOInst.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendEIODO.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendEIODODummy.h"
+#include "OpcUaPiXtendServer/PiXtend/PiXtendEIODOInst.h"
+
+using namespace OpcUaPiXtendServer;
+
+int main(int argc, char** argv)
+{
+
+    // initialize configuration modules
+
+    std::shared_ptr<PiXtendV2S> pixtendV2SSPtr = nullptr;
+#ifdef OPCUAPIXTENDSERVER_MODUL_V2S_DUMMY
+    pixtendV2SSPtr = std::make_shared<PiXtendV2SDummy>();
+#else
+    pixtendV2SSPtr = std::make_shared<PiXtendV2SInst>();
+#endif
+
+    std::shared_ptr<PiXtendV2L> pixtendV2LSPtr = nullptr;
+#ifdef OPCUAPIXTENDSERVER_MODUL_V2L_DUMMY
+    pixtendV2LSPtr = std::make_shared<PiXtendV2LDummy>();
+#else
+    pixtendV2LSPtr = std::make_shared<PiXtendV2LInst>();
+#endif
+
+    std::shared_ptr<PiXtendEIOAO> pixtendEIOAOSPtr = nullptr;
+#ifdef OPCUAPIXTENDSERVER_MODUL_V2L_DUMMY
+    pixtendEIOAOSPtr = std::make_shared<PiXtendEIOAODummy>();
+#else
+    pixtendEIOAOSPtr = std::make_shared<PiXtendEIOAOInst>();
+#endif
+
+    std::shared_ptr<PiXtendEIODO> pixtendEIODOSPtr = nullptr;
+#ifdef OPCUAPIXTENDSERVER_MODUL_V2L_DUMMY
+    pixtendEIODOSPtr = std::make_shared<PiXtendEIODODummy>();
+#else
+    pixtendEIODOSPtr = std::make_shared<PiXtendEIODOInst>();
+#endif
```

Comment:
This source code is also required in the opc ua server. Therefore a separate class is better for it.